File: src/scripts/fib.py

def fib(n):
    SZ = 2**64

    i = 0
    a, b = 1, 0
    while i < n:
        t = b
        b = (b + a) % SZ
        a = t
        i += 1

    return b

if __name__ == '__main__':
    print fib(100000)

File: src/lang/fib/fib.java

class fib
{
    public static void main(String[] args)
    {
        long n = Integer.parseInt(args[0]);
        long a = 1, b = 0, t;

        for(long i = 0; i < n; i++) {
            t = b;
            b = a + b;
            a = t;
        }

        System.out.printf("%d\n", b);
    }
}
